hera_qm/ant_metrics.py

```diff
@@ -0,0 +1,394 @@
+        return mean_Vij_metrics(self.data, pols, self.antpols, self.ants, self.bls, xants=xants, rawMetric=rawMetric)
+
+
+    def red_corr_metrics(self, pols=None, xants=[], rawMetric=False, crossPol=False):
+        '''Local wrapper for red_corr_metrics in hera_qm.ant_metrics module.'''
+
+        if pols is None:
+            pols = self.pols
+        return red_corr_metrics(self.data, pols, self.antpols, self.ants, self.reds, xants=xants, rawMetric=rawMetric, crossPol=crossPol)
+
+    def mean_Vij_cross_pol_metrics(self, xants=[], rawMetric=False):
+        '''Local wrapper for mean_Vij_cross_pol_metrics in hera_qm.ant_metrics module.'''
+        
+        return mean_Vij_cross_pol_metrics(self.data, self.pols, self.antpols, self.ants, self.bls, xants=xants, rawMetric=rawMetric)
+
+
+    def red_corr_cross_pol_metrics(self, xants=[], rawMetric=False):
+        '''Local wrapper for red_corr_cross_pol_metrics in hera_qm.ant_metrics module.'''
+
+        return red_corr_cross_pol_metrics(self.data, self.pols, self.antpols, self.ants, self.reds, xants=xants, rawMetric=False)
+    
+    def _run_all_metrics(self):
+        '''Designed to be run as part of AntennaMetrics.iterative_antenna_metrics_and_flagging().'''
+        
+        #Compute all raw metrics
+        meanVij = self.mean_Vij_metrics(xants=self.xants, rawMetric=True)
+        redCorr = self.red_corr_metrics(pols=['xx','yy'], xants=self.xants, rawMetric=True)
+        meanVijXPol = self.mean_Vij_cross_pol_metrics(xants=self.xants, rawMetric=True)
+        redCorrXPol = self.red_corr_cross_pol_metrics(xants=self.xants, rawMetric=True)   
+        
+        #Save all metrics and zscores
+        metrics, modzScores = {}, {}
+        for metName in ['meanVij','redCorr','meanVijXPol','redCorrXPol']:
+            metric = eval(metName)
+            metrics[metName] = metric
+            modz = per_antenna_modified_z_scores(metric)
+            modzScores[metName] = modz
+            for key in metric.keys():
+                if self.finalMetrics.has_key(metName):
+                    self.finalMetrics[metName][key] = metric[key]
+                    self.finalModzScores[metName][key] = modz[key]
+                else:
+                    self.finalMetrics[metName] = {key: metric[key]}
+                    self.finalModzScores[metName] = {key: modz[key]}
+        self.allMetrics.append(metrics)
+        self.allModzScores.append(modzScores)        
+    
+    def iterative_antenna_metrics_and_flagging(self, crossCut=5, deadCut=5, verbose=False):
+        '''Runs all four metrics (two for dead antennas two for cross-polarized antennas) and saves
+        the results internally to this this antenna metrics object. 
+        
+        Arguments:
+        crossCut -- Modified z-score cut for most cross-polarized antenna. Default 5 "sigmas".
+        deadCut -- Modified z-score cut for most likely dead antenna. Default 5 "sigmas".
+        '''
+        
+        #Summary statistics
```

Comment:
Since these summary statistics really only make sense in the context of running all four metrics iteratively, I'm OK with them only getting instantiated here. However, I do see some rationale in letting the user reset them manually, so I'll break that out.